plantseg/segmentation/dtws.py

```diff
@@ -1,8 +1,50 @@
+import time
+from functools import partial
+
 import numpy as np
-from GASP.segmentation import WatershedOnDistanceTransformFromAffinities
+from elf.segmentation.watershed import distance_transform_watershed, stacked_watershed
 
+from plantseg.pipeline import gui_logger
 from plantseg.pipeline.steps import AbstractSegmentationStep
-from plantseg.segmentation.utils import shift_affinities
+
+
+def compute_distance_transfrom_watershed(pmaps, threshold, sigma_seeds,
+                                         stacked=False, sigma_weights=2.,
+                                         min_size=100, alpha=1.0,
+                                         pixel_pitch=None, apply_nonmax_suppression=False,
+                                         n_threads=None):
+    """ Wrapper around elf.distance_transform_watershed
+
+    Args:
+        pmaps (np.ndarray): input height map.
+        threshold (float): value for the threshold applied before distance transform.
+        sigma_seeds (float): smoothing factor for the watershed seed map.
+        stacked (bool): if true the ws will be executed in 2D slice by slice, otherwise in 3D.
+        sigma_weights (float): smoothing factor for the watershed weight map (default: 2).
+        min_size (int): minimal size of watershed segments (default: 100)
+        alpha (float): alpha used to blend input_ and distance_transform in order to obtain the
+            watershed weight map (default: .9)
+        pixel_pitch (list-like[int]): anisotropy factor used to compute the distance transform (default: None)
+        apply_nonmax_suppression (bool): whether to apply non-maximum suppression to filter out seeds.
+            Needs nifty. (default: False)
+        n_threads (int): if not None, parallelize the 2D stacked ws. (default: None)
+
+    Returns:
+        np.ndarray: watershed segmentation
+    """
+
+    ws_kwargs = dict(threshold=threshold, sigma_seeds=sigma_seeds,
+                     sigma_weights=sigma_weights,
+                     min_size=min_size, alpha=alpha,
+                     pixel_pitch=pixel_pitch, apply_nonmax_suppression=apply_nonmax_suppression)
+    if stacked:
+        # WS in 2D
+        ws, _ = stacked_watershed(pmaps, ws_function=distance_transform_watershed, n_threads=n_threads, **ws_kwargs)
+    else:
+        # WS in 3D
+        ws, _ = distance_transform_watershed(pmaps, **ws_kwargs)
+
+    return ws
 
 
 class DistanceTransformWatershed(AbstractSegmentationStep):
```

Comment:
We may think about providing `pixel_pitch` argument if the voxel resolution is specified in the input. We should test it, if it makes a huge difference though. Not a big deal for now though

Reply:
Yes, I think that would be a good idea to add it.

Reply:
Sound good, I will test the `pixel_pitch` and let you know if it improves things.